src/main/java/com/redhat/devtools/intellij/commonuitest/fixtures/dialogs/settings/SettingsDialog.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.commonuitest.fixtures.dialogs.settings;
+
+import com.intellij.remoterobot.RemoteRobot;
+import com.intellij.remoterobot.data.RemoteComponent;
+import com.intellij.remoterobot.fixtures.CommonContainerFixture;
+import com.intellij.remoterobot.fixtures.DefaultXpath;
+import com.intellij.remoterobot.fixtures.FixtureName;
+import com.intellij.remoterobot.fixtures.JButtonFixture;
+import com.intellij.remoterobot.fixtures.JTreeFixture;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Settings dialog fixture
+ *
+ * @author zcervink@redhat.com
+ */
+@DefaultXpath(by = "MyDialog type", xpath = "//div[@class='MyDialog']")
+@FixtureName(name = "Settings Dialog")
+public class SettingsDialog extends CommonContainerFixture {
+    public SettingsDialog(@NotNull RemoteRobot remoteRobot, @NotNull RemoteComponent remoteComponent) {
+        super(remoteRobot, remoteComponent);
+    }
+
+    /**
+     * Navigate to specific 'Settings' dialog page according to given path
+     *
+     * @param path path to navigate to
+     */
+    public void navigateTo(String... path) {
+        if (path.length == 0) {
+            return;
```

Comment:
If you run that code, nothing happens. Since there is no homepage or no "the top node", this code simple does nothing. **Should I replace it with code that throws an exception?** Maybe it will make more sense than this.

Reply:
Absolutely. Either remove the condition so that the following code can handle the empty parameter or throw an exception. I would prefer the first mentioned.

Reply:
Fixed.